ui/src/components/SampleControls/SampleControls.jsx

```diff
@@ -76,96 +77,15 @@ class SampleControls extends React.Component {
   render() {
     const { hardwareObjects, getControlAvailability } = this.props;
 
-    const focusMotorProps = this.props.uiproperties.sample_view.components.find(
-      (c) => c.role === 'focus',
-    );
-
-    const zoomMotorProps = this.props.uiproperties.sample_view.components.find(
-      (c) => c.role === 'zoom',
-    );
-
-    const zoomMotor = this.props.hardwareObjects[zoomMotorProps.attribute];
-
     return (
       <div className={styles.controls}>
         {getControlAvailability('snapshot') && (
           <SnapshotControl canvas={this.props.canvas} />
         )}
         {getControlAvailability('draw_grid') && <GridControl />}
         {getControlAvailability('3_click_centring') && <CentringControl />}
-        {getControlAvailability('focus') && (
-          <OverlayTrigger
-            trigger="click"
-            rootClose
-            placement="bottom"
-            overlay={
-              <div className={styles.overlay}>
-                <OneAxisTranslationControl motorProps={focusMotorProps} />
-              </div>
-            }
-          >
-            <Button
-              className={styles.controlBtn}
-              name="focus"
-              title="Focus"
-              data-toggle="tooltip"
-            >
-              <i className={`${styles.controlIcon} fas fa-adjust`} />
-              <span className={styles.controlLabel}>Focus</span>
-            </Button>
-          </OverlayTrigger>
-        )}
-        {getControlAvailability('zoom') && (
-          <OverlayTrigger
-            trigger="click"
-            rootClose
-            placement="bottom"
-            overlay={
-              <div className={styles.overlay}>
-                <input
-                  className={styles.zoomSlider}
-                  type="range"
-                  min={0}
-                  max={zoomMotor.commands.length - 1}
-                  value={zoomMotor.commands.indexOf(zoomMotor.value)}
-                  disabled={zoomMotor.state !== HW_STATE.READY}
-                  onMouseUp={(e) => {
-                    this.props.setAttribute(
-                      'diffractometer.zoom',
-                      zoomMotor.commands[Number.parseFloat(e.target.value)],
-                    );
-                  }}
-                  onChange={(e) => {
-                    this.props.setBeamlineAttribute(
-                      'diffractometer.zoom',
-                      zoomMotor.commands[Number.parseFloat(e.target.value)],
-                    );
-                  }}
```

Comment:
We would make a local update to the store `onChange` and a server update `onMouseUp`. This led to a "jump" of the slider thumb when interacting with the mouse. More importantly, server updates were missed when interacting with the keyboard!

I kept only `onChange` and replaced `setBeamlineAttribute` with `setAttribute` to always (and only) perform server updates. The local state is still updated via websockets.